routers/api/v1/repo/label.go

```diff
@@ -231,6 +232,11 @@ func EditLabel(ctx *context.APIContext) {
 	if form.Description != nil {
 		l.Description = *form.Description
 	}
+	if form.IsArchived != nil && *form.IsArchived == true {
+		l.ArchivedUnix = timeutil.TimeStampNow()
+	} else {
+		l.ArchivedUnix = timeutil.TimeStamp(0)
+	}
```

Comment:
```suggestion
	l.Archived(form.IsArchived != nil && *form.IsArchived)
```
```suggestion
	l.SetArchived(form.IsArchived != nil && *form.IsArchived)
```
```suggestion
	l.SetArchived(form.IsArchived != nil && form.IsArchived)
```
Depending on which suggestions you apply.